### DISCUSSION TITLE
text sizing issue

### DISCUSSION THREAD

**Discussion by zsviczian** [excalidraw/excalidraw#3829] [Q&A]
It seems that with 0.9.0 the sizing of text has an issue. This involves only the Obsidian Android plugin. The exact same plugin code works without this issue on windows desktop and iOS. Also the same code works without issues with Excalidraw 0.8.0.

I can't even think where to start debugging this. Any suggestions?

https://user-images.githubusercontent.com/14358394/125273024-cc33ed80-e30c-11eb-8412-5752bffa75e0.mp4

**Comment by zsviczian** [comment]
I downgraded to 0.8.0 and the issue persists. So the problem is clearly not 0.9.0 related. Still not sure how to debug it, but I am getting closer...

**Reply by dwelle** [reply]
I assume you can't reproduce on excalidraw.com, correct? Is there anything custom you're doing in the plugin?

The only thing wrong I'm noticing in the video is the text gets larger after confirm which may account for that dimension difference.

**Reply by ad1992** [reply]
I don't see this issue in excalidraw.com, so probably related to the integration. 
@zsviczian any host CSS which could be affecting excalidraw texteditor?

**Reply by zsviczian** [reply]
Yes, the resizing is extremely strange. I tried turning off all plugin automation, the issue persisted on Android.

What should I be looking for in the css?

I have the font in the header of the document, which has a different link compared to the font referenced in the SVGs, but I wouldn't think that edit and display use different fonts.

![image](https://user-images.githubusercontent.com/14358394/125352596-c0bee180-e361-11eb-8d32-c7fb2c1c767b.png)

![image](https://user-images.githubusercontent.com/14358394/125352719-e9df7200-e361-11eb-8ef8-e7c8f8c6014c.png)

**Reply by dwelle** [reply]
I wouldn't say it's necessarily a CSS thing, rather something with how we compute the text dimensions and/or logic around the text editor.

**Reply by zsviczian** [reply]
I tested Excalidraw Obsidian on multiple Android devices. Interestingly I only experience the issue on my Galaxy Note 8. Must be a compatibility issue... though it is not clear how and why... Also, excalidraw.com on the same device does not demonstrate the same behavior.

**Reply by zsviczian** [reply]
Here's the same drawing... from Excalidraw Obsidian on a desktop and on a mobile. Seems the text is smaller and the baseline is off.

![image](https://user-images.githubusercontent.com/14358394/125510025-f4c23f38-9bed-4ded-a06f-96b9fb1b9e59.png)

![image](https://user-images.githubusercontent.com/14358394/125510293-c95abc6a-0f3a-4984-879e-a154243cbde2.png)

**Reply by zsviczian** [reply]
I am testing with "20px Virgil". In both cases document.fonts.check("20px Virgil") returns true. 
But in the correct case {witdth: 144, height: 25, baseline:18} and in the truncated case it is {width:130, height:23, baseline 16}

**Reply by zsviczian** [reply]
So... I have a solution - sort of.
Until now I've been using my device in WQHD+ mode. When I changed the resolution to HD+ the error went away.  
I still don't understand what the difference is between Obsidian and Excalidraw.com, since they both run on the same Chrome browser engine. But Excalidraw.com works with all the resolutions, and the component embedded in Obsidian does not.

![image](https://user-images.githubusercontent.com/14358394/125520574-965d2318-da6f-4a25-82d5-fa9f9c0f4c84.png)

**Reply by zsviczian** [reply]
my mistake. The screen resolution didn't resolve the issue, only decreased the inaccuracy. With longer text, the cutoff is still there.

**Comment by jakub-wojciech** [comment]
Is there any workaround for this issue? I have this on Android and I have no idea how to fix it. It makes it hard to use on mobile.

**Reply by zsviczian** [reply]
I no longer experience this issue (on the same devices as before). I don't recall at what point the issue got resolved.

Are you experiencing it? What device are you using? I assume you are and Obsidian Excalidraw user?

**Reply by jakub-wojciech** [reply]
I do experience that with Samsung S22 Ultra on the latest version of Obsidian Excalidraw (1.8.14).
If you don't experience that, I will try to play with settings maybe or reinstall Obsidian Excalidraw.
I will post an update if I find anything.

**Reply by jakub-wojciech** [reply]
- I tried to change themes but didn't fix the issue
- I did try to create a new vault on mobile with just Obsidian Excalidraw as an add-on, didn't fix the issue
- I have uninstalled and installed Obsidian again (settings weren't removed), didn't fix the issue

I am not sure what else I can do. What I see is that if I am writing the text on mobile, the size of the text is a bit smaller in edit mode; when I click out from the text, the text changes to be a bit bigger, and with longer messages, the last letters are lost. Also, when changing alignment to the right, the first letters are lost, and when middle alignment first and last letters are lost.
Not sure if this is related, but when I double-click on a text box, the text box changes position and sometimes goes out of the screen. It seems like it's always going to the same place when editing.
I can make a video if it can be helpful. I will try to play with it a little bit more and see if I can find anything.

**Reply by jakub-wojciech** [reply]
I just realized this is an "Excalidraw" repo, not "Obsidian Excalidraw"  😅

**Comment by annettevogt** [comment]
Hello everyone, I have some similar issues.

https://github.com/excalidraw/excalidraw/assets/65187965/b499891d-8aab-4c23-8ca2-cc9b5df55450

**Reply by annettevogt** [reply]
I had to bring the solution of the video down to less then 10MB. Sorry for the bad quality.

I am using a Samsung Galaxy S6 lite tablet with a Samsung pen.

**Comment by Daniel21b** [comment]
Still, experiencing the same issue? any updates
